File: app/src/main/java/test/juyoufuli/com/myapplication/mvp/ui/home/adapter/ArticleAdapter.java
package test.juyoufuli.com.myapplication.mvp.ui.home.adapter;

import android.view.View;

import com.jess.arms.base.BaseHolder;
import com.jess.arms.base.DefaultAdapter;

import java.util.List;

import javax.inject.Inject;

import test.juyoufuli.com.myapplication.R;
import test.juyoufuli.com.myapplication.mvp.entity.ArticleBean;

/**
 * Author : ludf
 * Created Time : 2018-09-27  14:57
 * Description:
 */
public class ArticleAdapter extends DefaultAdapter<ArticleBean> {
    private ArticleItemHolder.ChildClickListener mChildClickListener;

    @Inject
    public ArticleAdapter(List<ArticleBean> infos) {
        super(infos);
    }

    @Override
    public BaseHolder<ArticleBean> getHolder(View v, int viewType) {
        ArticleItemHolder articleItemHolder = new ArticleItemHolder(v, mChildClickListener);
        return articleItemHolder;
    }

    @Override
    public int getLayoutId(int viewType) {
        return R.layout.article_item;
    }

    public void setChildClickListener(ArticleItemHolder.ChildClickListener mChildClickListener) {
        this.mChildClickListener = mChildClickListener;
    }
}

File: app/src/main/java/test/juyoufuli/com/myapplication/mvp/ui/home/adapter/ArticleAdapter.kt
package test.juyoufuli.com.myapplication.mvp.ui.home.adapter

import android.view.View

import com.jess.arms.base.BaseHolder
import com.jess.arms.base.DefaultAdapter

import javax.inject.Inject

import test.juyoufuli.com.myapplication.R
import test.juyoufuli.com.myapplication.mvp.entity.ArticleBean

/**
 * Author : ludf
 * Created Time : 2018-09-27  14:57
 * Description:
 */
class ArticleAdapter @Inject
constructor(infos: List<ArticleBean>) : DefaultAdapter<ArticleBean>(infos) {
    private var mChildClickListener: ArticleItemHolder.ChildClickListener? = null

    override fun getHolder(v: View, viewType: Int): BaseHolder<ArticleBean> {
        return ArticleItemHolder(
                v, mChildClickListener!!
        )
    }

    override fun getLayoutId(viewType: Int): Int {
        return R.layout.article_item
    }

    fun setChildClickListener(mChildClickListener: ArticleItemHolder.ChildClickListener) {
        this.mChildClickListener = mChildClickListener
    }
}
